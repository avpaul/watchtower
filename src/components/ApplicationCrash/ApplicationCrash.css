.application-crash {
  min-height: 100vh;
  background-color: #f4f8f9;
}
.application-crash .app-icon-area {
  padding-top: 100px;
}
.application-crash .refresh-btn {
  border-radius: 5px;
  background-color: #10a36d;
  box-shadow: 0 2px 3px 0 rgba(0, 0, 0, 0.11);
  color: #ffffff;
  font-family: 'DIN Pro';
  font-size: 14px;
  font-weight: bold;
  letter-spacing: 0.58px;
  line-height: 18px;
  padding: 10px;
}
