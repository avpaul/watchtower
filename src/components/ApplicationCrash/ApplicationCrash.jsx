import React, { Fragment } from 'react';
import icon from '../../static/undraw_notify_88a4_ani_new.svg';
import './ApplicationCrash.scss';

const ApplicationCrash = () => (
  <Fragment>
    <div className="application-crash">
      <div className="container">
        <div className="row app-icon-area">
          <div className="col-md-12 text-center">
            <img src={icon} alt="icon" className="w-50" />
          </div>
        </div>
        <div className="row mt-5 mb-3 justify-content-center">
          <div className="col-md-8 text-center">
            <h4 className="font-weight-lighter">Oops! Something went wrong.</h4>
          </div>
        </div>
        <div className="row justify-content-center">
          <div className="col-md-12 text-center">
            <button
              type="button"
              className="refresh-btn text-uppercase"
              onClick={() => window.location.reload()}
            >
              Refresh
            </button>
          </div>
        </div>
      </div>
    </div>
  </Fragment>
);

export default ApplicationCrash;
