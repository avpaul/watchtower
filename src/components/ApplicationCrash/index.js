import ApplicationCrash from './ApplicationCrash';

export default ApplicationCrash;
