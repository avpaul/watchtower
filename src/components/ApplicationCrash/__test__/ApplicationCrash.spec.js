import React from 'react';
import { shallow } from 'enzyme';
import ApplicationCrash from '../ApplicationCrash';

describe('ApplicationCrash component', () => {
  it('it should render without crashing', () => {
    const wrapper = shallow(<ApplicationCrash />);
    wrapper.find('button').simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
});
