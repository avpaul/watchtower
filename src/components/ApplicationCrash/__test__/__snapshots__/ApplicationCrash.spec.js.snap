// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationCrash component it should render without crashing 1`] = `
<Fragment>
  <div
    className="application-crash"
  >
    <div
      className="container"
    >
      <div
        className="row app-icon-area"
      >
        <div
          className="col-md-12 text-center"
        >
          <img
            alt="icon"
            className="w-50"
            src="undraw_notify_88a4_ani_new.svg"
          />
        </div>
      </div>
      <div
        className="row mt-5 mb-3 justify-content-center"
      >
        <div
          className="col-md-8 text-center"
        >
          <h4
            className="font-weight-lighter"
          >
            Oops! Something went wrong.
          </h4>
        </div>
      </div>
      <div
        className="row justify-content-center"
      >
        <div
          className="col-md-12 text-center"
        >
          <button
            className="refresh-btn text-uppercase"
            onClick={[Function]}
            type="button"
          >
            Refresh
          </button>
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;
